cmd/kuberhealthy/config.go

```diff
@@ -154,8 +154,8 @@ func startConfigReloadMonitoringWithSmoothing(filePath string, scrapeInterval ti
 	return outChan, cancelFunc, nil
 }
 
-// configReloader watchers for events in file and restarts kuberhealhty checks
-func configReloader(ctx context.Context, kh *Kuberhealthy) {
+// configReloadNotifier watchers for events in file, reloads the configuration, and notifies upstream to restart checks
```

Comment:
This is really good, but it looks like `configReloadNotifier` will run forever.  This means that every time `kuberhealthy.Start()` is called, a new go routine will be spawned that never shuts down.  That is a very uncommon thing, but not impossible.

can you add a context here, then pass in the context when this routine is spawned?  Then, check if the context has expired and `return` if it has.